### ISSUE TITLE
Question son where appium should be running in hive ci setup

### ISSUE THREAD

**Issue by ambirag** [bbc/hive-scheduler#147]
Hi
Sorry this is the only avenue available to communicate with the team regarding doubts on Hive CI.
I have setup it up locally on my Mac and it successfully ran a github based UI automation on an iPhone and Android mobile devices connected to my Mac. In this case, schedule and runner were in the same box, Appium was running as well. test script was pointing to http://0.0.0.0:4723/wd/hub and ran fine.

Second Case:
I am also able to start a runner on another Mac(say mac2) and point to my schedule's private IP on Mac1 and I could see the iPhone attached to Mac2 via http://localhost:3000/workers on Mac1 (scheduler)
My questions:
- In the second case, when I run a test via "test batch", do  I have to run Appium on runner or scheduler?
- also I believe that if I have more than 1 runner and if i want my test to run on all of them, i don't have to have any parallelization built into the test. Just use env variables to run the same test pointing to different Appium servers?

**Comment by Asimk21** [comment]
Hi @ambirag,

1. There should be only one runner running on one Mac/linux machine. Depending on runner specified (like android or ios) it will detect the devices connected to machine. We can check which runner is configured and devices list and queue names by running `hived status`. [Refer](https://github.com/bbc/hive-runner)
2. Regarding how to start appium server in hive-ci. It is done through scripts. Hive starts it in different process. How to do it in hive? [Refer](http://bbc.github.io/hive-ci/documentation/appium_test.html)
3. Batch is the trigger point for test. When we specify queue name, build and other details in batch and submit, it should start tests on device associated with queue. 
4. [Refer](http://bbc.github.io/hive-ci/documentation/running-your-tests.html) this to create script, project and batch in hive.

At very high level:
`Script` runs on command line. It should include any pre-execution setup like starting appium node and execution command.
`Project` includes script and other information like queue name
`Batch` is trigger point for jobs/tasks/tests.

**Comment by Asimk21** [comment]
Just to add on to that we don't have to do any parallellization as that is handled by hive in different process. Just need to assign queue name for device.

**Comment by ambirag** [comment]
thanks @Asimk21 , my main question was where Appium should be running, as part of machine where runner is running is what I find.
`scheduler <-> runner 1 (runner + xcode + appium,etc)`
                       `      <-> runner 2 (runner + xcode + appium,etc) `

**Comment by Asimk21** [comment]
Yes, it should be running where runner is running. Any testing tool we use, is independent of Hive-Scheduler (its web interface to create and schedule jobs)
